core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -149,15 +149,13 @@ func (s *onchainSubscriptions) queryLoop() {
 		}
```

Comment:
The double if statement is a little confusing. Maybe you could place the second if statement within the first after the call to `updateLastKnownCount()`?